### PR TITLE
Made regenerate_organs() reset external organs as well.

### PR DESCRIPTION
**Pull request by lizardqueenlexi**
## About The Pull Request

As the title says, augments the species.regenerate_organs() proc to regenerate external organs. It appears this was missed when mutant organs were largely replaced with external organs - leading to cases where revival such as admin healing does _not_ currently restore external organs. This has been corrected.

As external organs can be placed on species that do not normally have them, regenerate_organs() will also check all external organs on the target carbon and remove ones that their species doesn't naturally have - properly restoring the "default" appearance of the mob.

Also slightly modifies on_species_gain() so that it does not insert external organs that are already there - which in some cases alongside a regenerate_organs() call, could cause a full extra set of external organs to drop onto the ground.

## Why It's Good For The Game

This proc is supposed to restore _all_ organs to default condition. In the current code, even admin-healing cannot restore external organs such as a moth's wings. Obviously, this is not intentional behavior.

## Changelog

:cl:
fix: made regenerate_organs() apply to external organs as well as internal ones
/:cl:

### CODE REVIEW

**Review by Kapu1178 — APPROVED** [review]
Probably good

File: code/modules/mob/living/carbon/human/species.dm [hunk 1]
```diff
@@ -402,8 +402,20 @@ GLOBAL_LIST_EMPTY(features_by_species)
...
 				QDEL_NULL(I)
-
-	for(var/organ_path in mutant_organs)
+	for(var/external_organ in C.external_organs)
+		// External organ checking. We need to check the external organs owned by the carbon itself,
+		// because we want to also remove ones not shared by its species.
+		// This should be done even if species was not changed.
+		if(external_organ in external_organs)
+			continue // Don't remove external organs this species is supposed to have.
+		var/obj/item/organ/I = C.getorgan(external_organ)
```
Comment (SuperNovaa41) [inline code/modules/mob/living/carbon/human/species.dm:411]: use a better variable name here, `I` is not good enough

File: code/modules/mob/living/carbon/human/species.dm [hunk 2]
```diff
@@ -470,7 +482,7 @@ GLOBAL_LIST_EMPTY(features_by_species)
 		for(var/obj/item/organ/external/organ_path as anything in external_organs)
 			//Load a persons preferences from DNA
 			var/obj/item/organ/external/new_organ = SSwardrobe.provide_type(organ_path)
-			new_organ.Insert(human)
+			new_organ.Insert(human, TRUE, FALSE)
```
Comment (SuperNovaa41) [inline code/modules/mob/living/carbon/human/species.dm:485]: use named args here

**Review by SuperNovaa41 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/species.dm (hunk 1 above)
Comment (lizardqueenlexi) [reply code/modules/mob/living/carbon/human/species.dm]: I have changed this to "current_organ" both in this and the section that handles mutant organs.

File: code/modules/mob/living/carbon/human/species.dm (hunk 2 above)
Comment (lizardqueenlexi) [reply code/modules/mob/living/carbon/human/species.dm]: I've now done this both here and in several other places in this file.

**Pull request merged**